frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -302,3 +304,87 @@ export class TransformPodData {
     return dcPodsData;
   }
 }
+
+const getOwnedResources = <T extends K8sResourceKind>(
+  { metadata: { uid } }: K8sResourceKind,
+  resources: T[],
+): T[] => {
+  return _.filter(resources, ({ metadata: { ownerReferences } }) => {
+    return _.some(ownerReferences, {
+      uid,
+      controller: true,
+    });
+  });
+};
+
+const getPodData = (
+  data: PodData,
+): { inProgressDeploymentData: Pod[] | null; completedDeploymentData: Pod[] | null } => {
+  // Scaling no. of pods
+  if (data.currentRCPhase === 'Complete' && data.currentAvailableReplicas !== data.totalReplicas) {
+    return { inProgressDeploymentData: null, completedDeploymentData: data.current };
+  }
+
+  // Deploy
+  switch (data.strategy) {
+    case 'Recreate':
+      if (
+        data.previous.length === 0 &&
+        data.currentReplicas === data.totalReplicas &&
+        data.currentAvailableReplicas === data.totalReplicas
+      ) {
+        return { inProgressDeploymentData: null, completedDeploymentData: data.current };
+      }
+      return { inProgressDeploymentData: data.current, completedDeploymentData: data.previous };
+    default:
+      // Rolling
+      if (data.previous.length > 0) {
+        return { inProgressDeploymentData: data.current, completedDeploymentData: data.previous };
+      }
+      return { inProgressDeploymentData: null, completedDeploymentData: data.current };
+  }
+};
+
+export const rollback = (dc: K8sResourceKind, previous): Promise<K8sResourceKind> => {
+  const req = {
+    kind: 'DeploymentConfigRollback',
+    apiVersion: 'apps.openshift.io/v1',
+    spec: {
+      from: {
+        name: _.get(previous, 'metadata.name'),
+      },
+      includeTriggers: false,
+      includeTemplate: true,
+      includeReplicationMeta: false,
+      includeStrategy: true,
+    },
+    name: dc.metadata.name,
+    latest: true,
+    force: true,
+  };
+  const opts = {
+    name: dc.metadata.name,
+    ns: dc.metadata.namespace,
+    path: 'rollback',
+  };
+  return k8sCreate(DeploymentConfigModel, req, opts);
+};
+
+export const getDataBasedOnCurrentAndPreviousRC = (
+  d: K8sResourceKind,
+  current,
+  previous,
+  pods: Pod[],
+) => {
+  const currentPods = current && current.metadata ? getOwnedResources(current, pods) : [];
+  const previousPods = previous && previous.metadata ? getOwnedResources(previous, pods) : [];
+  return getPodData({
+    current: [...currentPods],
+    previous: [...previousPods],
```

Comment:
Same as above.

Reply:
fixed in https://github.com/openshift/console/pull/2456/commits/d069d28b40bb578d9a26810bb61269647f5d241b